rfc/0006-threat-model.md

```diff
@@ -0,0 +1,262 @@
+for any reasion, API server should reject the request. We are safe on this.
+Kubewarden default behavior  already does that.
+
+## Threat 2 - Attacker passes workloads which require complex  processing causing timeouts
+
+An attacker, who can access the admission controller at a network level, passes
+requests to  the admission controller which require complex processing, causing
+timeouts as the  admission controller uses compute power to process the workloads
+
+**Mitigation**
+Webhook fail closed and authenticate callers. We are safe, Kubewarden already does that.
+
+## Threat 3 - Attacker exploits misconfiguration of webhook to bypass
+An attacker, who has rights to create workloads in the cluster, is able to exploit
+a mis-configuration to bypass the intended security control
+
+**Mitigation**
+Regular reviews of webhook configuration catch issues.
+
+**To do**
+We may provide some mechanism to notify operators when a webhook is changed.
+Maybe the controller can watch its webhooks, fix when the configuration is
+changed and notify the operator.
+
+### Threat 4 - Attacker has rights to delete or modify the k8s webhook  object
+
+An attacker who has Kubernetes API access, has sufficient privileges to delete
+the webhook  object in the cluster.
+
+**Mitigation**
+RBAC rights are strictly controlled.
+
+**To do**
+I think most of the RBAC is not Kubewarden responsibility. But we can help our users if we:
+- Warning them in our docs and *suggest* some minimum RBAC to be used.
+- Provide a policy which detect RBAC changes and **maybe** block them.
+
+
+## Threat 5 - Attacker gets access to valid credentials for the webhook
+An attacker gains access to valid client credentials for the admission controller webhook
+
+**Mitigation**
+Webhook fails closed.
+
+Kubewarden is failed closed. Thus, we should be fine.
+
+## Threat  6 - Attacker gains access to a cluster admin credential
+
+An attacker gains access to a cluster-admin level credential in the kubernetes cluster.
+
+**Mitigation**
+N/A
+
+I cannot see how Kubewarden can help here.
+
+## Threat 7 - Attacker sniffs traffic on the container network
+An attacker who has access to the container network is able to sniff traffic
+between the API  server and the admission controller webhook.
+
+**Mitigation**
+Webhook uses TLS encryption for all traffic
+
+Kubewarden are safe. Because the webhook connections are encrypted.
+
+
+## Threat 8 - Attacker carries out a MITM attack on the webhook
+An attacker on the container network, who has access to the NET_RAW capability
+can  attempt to use MITM tooling to intercept traffic between the API server
+and admission  controller webhook.
+
+**Mitigation**
+Webhook mTLS authentication is used.
+
+**To do**
+Kubewarden should implement mutual TLS authentication
+We can add in the recommended policies from the `kubewarden-defaults` Helm
+chart a policy to drop the `NET_RAW` capability.
```

Comment:
`NET_RAW` containers will always exist though. CNI plugins are a typical example.

Reply:
I suppose then drop NET_RAW capability besides in a rejectlist ns (control plane).

Reply:
We can prevent the creation of containers that have this capability via [this policy](https://github.com/kubewarden/capabilities-psp-policy). Is that one of the policies we enable via our "PSP best practices" selection?